--- origsrc/unrar/makefile	2017-06-13 06:04:49.000000000 -0500
+++ src/unrar/makefile	2017-09-04 03:54:45.149132900 -0500
@@ -154,9 +154,9 @@ sfx:	clean $(OBJECTS)
 lib:	WHAT=RARDLL
 lib:	CXXFLAGS+=$(LIBFLAGS)
 lib:	clean $(OBJECTS) $(LIB_OBJ)
-	@rm -f libunrar.so
+	@rm -f cygunrar.dll libunrar.dll.a
 	@rm -f libunrar.a
-	$(LINK) -shared -o libunrar.so $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
+	$(LINK) -shared -Wl,--out-implib,libunrar.dll.a -o cygunrar.dll $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
 	$(AR) rcs libunrar.a $(OBJECTS) $(LIB_OBJ)
 
 install-unrar:
@@ -166,7 +166,9 @@ uninstall-unrar:
 			rm -f $(DESTDIR)/bin/unrar
 
 install-lib:
-		install libunrar.so $(DESTDIR)/lib
+		install -D dll.hpp $(DESTDIR)/include/unrar/dll.hpp
+		install -D cygunrar.dll $(DESTDIR)/bin/cygunrar.dll
+		install -D libunrar.dll.a $(DESTDIR)/lib/libunrar.dll.a
 		install libunrar.a $(DESTDIR)/lib
 
 uninstall-lib:
